pgjdbc/src/test/java/org/postgresql/test/jdbc4/jdbc41/SchemaTest.java

```diff
@@ -217,6 +223,98 @@ public void testSearchPathPreparedStatement() throws SQLException {
+      assertThat("Test creates function in schema 'schema2' and this function try use table \"check_table\" "
+          + "from schema 'schema1'. We expect here sql error code - " + PSQLState.RELATION_DOES_NOT_EXIST + ", because "
+          + " search_path does not contains schema 'schema1' and postgres does not see table \"check_table\"",
+        sqlState,
+        equalTo(PSQLState.RELATION_DOES_NOT_EXIST.getState())
+      );
+      assertThat(
+        "Test creates function in schema 'schema2' and this function try use table \"check_table\" "
+          + "from schema 'schema1'. We expect here that sql error message will be contains \"check_table\", because "
+          + " search_path does not contains schema 'schema1' and postgres does not see table \"check_table\"",
+        message,
+        containsString("\"check_table\"")
+      );
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail("Unexpected exception is occurred " + e.getMessage());
+    }
+    connection.close();
+  }
+
+  @Test
+  public void testCurrentSchemaPropertyVisibilityFunction() throws SQLException {
+    testCurrentSchemaPropertyVisibilityTableDuringFunctionCreation();
+    Properties properties = new Properties();
+    properties.setProperty(PGProperty.CURRENT_SCHEMA.getName(), "public,schema1,schema2");
+    Connection connection = TestUtil.openDB(properties);
+
+    TestUtil.execute("select check_fun()", connection);
+    connection.close();
+  }
+
+  @Test
+  public void testCurrentSchemaPropertyNotVisibilityTableInsideFunction() throws SQLException {
+    testCurrentSchemaPropertyVisibilityTableDuringFunctionCreation();
+    Properties properties = new Properties();
+    properties.setProperty(PGProperty.CURRENT_SCHEMA.getName(), "public,schema2");
+    Connection connection = TestUtil.openDB(properties);
+
+    try {
+      TestUtil.execute("select check_fun()", connection);
+    } catch (PSQLException e) {
+      String sqlState = e.getSQLState();
+      String message = e.getMessage();
+      assertThat("Test call function in schema 'schema2' and this function uses table \"check_table\" "
+          + "from schema 'schema1'. We expect here sql error code - " + PSQLState.RELATION_DOES_NOT_EXIST + ", because "
+          + " search_path does not contains schema 'schema1' and postgres does not see table \"check_table\".",
+        sqlState,
+        equalTo(PSQLState.RELATION_DOES_NOT_EXIST.getState())
+      );
+      assertThat(
+        "Test call function in schema 'schema2' and this function uses table \"check_table\" "
+          + "from schema 'schema1'. We expect here that sql error message will be contains \"check_table\", because "
+          + " search_path does not contains schema 'schema1' and postgres does not see table \"check_table\"",
+        message,
+        containsString("\"check_table\"")
+      );
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail("Unexpected exception is occurred " + e.getMessage());
```

Comment:
This does not seem to add much.
Those exceptions could be just re-thrown.

Reply:
fixed